### PR TITLE
CWL input file format validation. Closes #3569

### LINKED ISSUE #3569: CWL: Validate input input files against OWL/RDF
A/C:
- centaur test that validates that inputs schemas are used for validation when the input specifies a format
- centaur test that validates that inputs schemas are ignored for validation when the input doesn't specify a format

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: centaurCwlRunner/src/main/scala/centaur/cwl/OutputManipulator.scala [hunk 1]
```diff
@@ -173,20 +204,20 @@ object OutputManipulator extends Poly1 {
...
+            case (k, v) => k -> resolveOutputViaInnerType(typeMap(k))(cwl, v, pathBuilder)
           }).asJson
         } yield outputJson).getOrElse(throw new RuntimeException(s"We currently do not support output record schemas with ${tpe.select[OutputArraySchema].get.items} inner type"))
       case (JsNull, Inl(CwlType.Null)) => Json.Null
       case (json, tpe) => throw new RuntimeException(s"We currently do not support outputs (${json.getClass.getSimpleName}) of $json and type $tpe")
     }
   }
 
-  implicit def moit: Case.Aux[MyriadOutputInnerType, (JsValue, PathBuilder) => Json] = at[MyriadOutputInnerType] {
+  implicit val moit: Case.Aux[MyriadOutputInnerType, (Cwl, JsValue, PathBuilder) => Json] = at {
```
[the remark this answers is not in the document]
Comment (kshakir) [reply centaurCwlRunner/src/main/scala/centaur/cwl/OutputManipulator.scala]: I think they're `def`s in the docs. But everywhere I've seen them in real code they've been `val`s. No clue what the compile-time/runtime CPU & memory tradeoffs are.

File: cwl/src/main/scala/cwl/ontology/Schema.scala [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    } finally {
+      schemaReasoner.dispose()
+    }
+  }
+
+  private val ontologyManager: OWLOntologyManager = OWLManager.createOWLOntologyManager
+  private val dataFactory: OWLDataFactory = ontologyManager.getOWLDataFactory
+  private val reasonerFactory: OWLReasonerFactory = new StructuralReasonerFactory
+  private val schemaOntology: OWLOntology = ontologyManager.createOntology()
+  private val schemaPrefixManager: PrefixManager =
```
Comment (danbills) [inline cwl/src/main/scala/cwl/ontology/Schema.scala]: can `OWLOntologyManager` and `OWLReasonerFactory` be constructor params w/ defaults?

File: project/Dependencies.scala [hunk 3]
```diff
@@ -313,6 +314,29 @@ object Dependencies {
...
+  jcl-over-slf4j.jar is a replacement for commons-logging 1.1.1. Meanwhile our extensive transitive use of apache's
+  httpclient has been including commons-logging 1.2 for a while. Now the owl api dependency jcl-over-slf4j is
+  conflicting during assembly. As there have been no reported errors AFAIK with commons-logging leaving it in for now.
+  However as we use slf4j for cromwell log configuration the correct thing might actually be to exclude commons-logging
+  whenever importing httpclient and include jcl-over-slf4j. That way we can control all of our logging in one place.
+
+  - https://www.slf4j.org/legacy.html#jclOverSLF4J
+   */
+  val owlApiDependencies = List(
+    "net.sourceforge.owlapi" % "owlapi-distribution" % owlApiV
```
Comment (danbills) [inline project/Dependencies.scala]: nice find!

File: cwl/src/main/scala/cwl/ontology/Schema.scala (hunk 2 above)
Comment (kshakir) [reply cwl/src/main/scala/cwl/ontology/Schema.scala]: Exposed! Though no-one is using those parameters afaict & in general I like to encapsulate APIs.

I'm really hoping **no one** touches the `owl-api` directly outside of this package. Depending on what happens in the future we may need to switch libraries, like we have many other times:
- `spray` ➔ `akka-http`
- `spray-json` ➔ `circe`
- `nashorn` ➔ `rhino`

For example, there are _still_ `.ask()` calls left over after Akka abandoned `spray` and we had to (mostly) upgrade.

File: project/Dependencies.scala (hunk 3 above)
Comment (kshakir) [reply project/Dependencies.scala]: h/t: @rushtong and @davidangb. I think OWL/RDF may be in the running vs JSON for the number of JVM library [options](https://xkcd.com/927/).

**Pull request merged**